### PR TITLE
feat(plasma-tokens-native): Add generate complex gradient for iOS

### PR DESCRIPTION
**Pull request by neretin-trike**
:baby_chick: Download canary assets:

[color_sberHealth_ios-swift--canary.548.5185739481.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.548.5185739481.swift)  
[color_sberHealth_kotlin--canary.548.5185739481.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.548.5185739481.kt)  
[color_sberHealth_react-native--canary.548.5185739481.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.548.5185739481.ts)  
[color_sbermarket_business_ios-swift--canary.548.5185739481.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.548.5185739481.swift)  
[color_sbermarket_business_kotlin--canary.548.5185739481.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.548.5185739481.kt)  
[color_sbermarket_business_react-native--canary.548.5185739481.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.548.5185739481.ts)  
[color_sbermarket_ios-swift--canary.548.5185739481.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.548.5185739481.swift)  
[color_sbermarket_kotlin--canary.548.5185739481.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.548.5185739481.kt)  
[color_sbermarket_metro_ios-swift--canary.548.5185739481.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.548.5185739481.swift)  
[color_sbermarket_metro_kotlin--canary.548.5185739481.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.548.5185739481.kt)  
[color_sbermarket_metro_react-native--canary.548.5185739481.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.548.5185739481.ts)  
[color_sbermarket_react-native--canary.548.5185739481.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.548.5185739481.ts)  
[color_sbermarket_selgros_ios-swift--canary.548.5185739481.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.548.5185739481.swift)  
[color_sbermarket_selgros_kotlin--canary.548.5185739481.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.548.5185739481.kt)  
[color_sbermarket_selgros_react-native--canary.548.5185739481.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.548.5185739481.ts)  
[color_sbermarket_wlbusiness_ios-swift--canary.548.5185739481.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_ios-swift--canary.548.5185739481.swift)  
[color_sbermarket_wlbusiness_kotlin--canary.548.5185739481.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_kotlin--canary.548.5185739481.kt)  
[color_sbermarket_wlbusiness_react-native--canary.548.5185739481.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_react-native--canary.548.5185739481.ts)  
[color_sberonline_ios-swift--canary.548.5185739481.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_ios-swift--canary.548.5185739481.swift)  
[color_sberonline_kotlin--canary.548.5185739481.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_kotlin--canary.548.5185739481.kt)  
[color_sberonline_react-native--canary.548.5185739481.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberonline_react-native--canary.548.5185739481.ts)  
[color_sberprime_ios-swift--canary.548.5185739481.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.548.5185739481.swift)  
[color_sberprime_kotlin--canary.548.5185739481.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.548.5185739481.kt)  
[color_sberprime_react-native--canary.548.5185739481.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.548.5185739481.ts)  
[shadow_sbermarket_react-native--canary.548.5185739481.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.548.5185739481.ts)  
[typo_mage_ios-swift--canary.548.5185739481.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.548.5185739481.swift)  
[typo_mage_kotlin--canary.548.5185739481.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.548.5185739481.kt)  
[typo_mage_react-native--canary.548.5185739481.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.548.5185739481.ts)  
[typo_plasma_ios-swift--canary.548.5185739481.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.548.5185739481.swift)  
[typo_plasma_kotlin--canary.548.5185739481.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.548.5185739481.kt)  
[typo_plasma_react-native--canary.548.5185739481.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.548.5185739481.ts)  
[typo_ruler_ios-swift--canary.548.5185739481.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.548.5185739481.swift)  
[typo_ruler_kotlin--canary.548.5185739481.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.548.5185739481.kt)  
[typo_ruler_react-native--canary.548.5185739481.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.548.5185739481.ts)  
[typo_sage_ios-swift--canary.548.5185739481.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.548.5185739481.swift)  
[typo_sage_kotlin--canary.548.5185739481.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.548.5185739481.kt)  
[typo_sage_react-native--canary.548.5185739481.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.548.5185739481.ts)  
[typo_sbermarket_ios-swift--canary.548.5185739481.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.548.5185739481.swift)  
[typo_sbermarket_kotlin--canary.548.5185739481.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.548.5185739481.kt)  
[typo_sbermarket_react-native--canary.548.5185739481.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.548.5185739481.ts)  
[typo_soulmate_ios-swift--canary.548.5185739481.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.548.5185739481.swift)  
[typo_soulmate_kotlin--canary.548.5185739481.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.548.5185739481.kt)  
[typo_soulmate_react-native--canary.548.5185739481.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.548.5185739481.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.217.0-canary.548.5185739481.0
  npm install @salutejs/plasma-core@1.122.0-canary.548.5185739481.0
  npm install @salutejs/plasma-hope@1.217.0-canary.548.5185739481.0
  npm install @salutejs/plasma-icons@1.151.0-canary.548.5185739481.0
  npm install @salutejs/plasma-temple@1.166.0-canary.548.5185739481.0
  npm install @salutejs/plasma-tokens-b2b@1.22.0-canary.548.5185739481.0
  npm install @salutejs/plasma-tokens-b2c@0.31.0-canary.548.5185739481.0
  npm install @salutejs/plasma-tokens-web@1.37.0-canary.548.5185739481.0
  npm install @salutejs/plasma-tokens@1.56.0-canary.548.5185739481.0
  npm install @salutejs/plasma-ui@1.198.0-canary.548.5185739481.0
  npm install @salutejs/plasma-web@1.217.0-canary.548.5185739481.0
  npm install @salutejs/plasma-cy-utils@0.63.0-canary.548.5185739481.0
  npm install @salutejs/plasma-sb-utils@0.120.0-canary.548.5185739481.0
  npm install @salutejs/plasma-tokens-utils@0.27.0-canary.548.5185739481.0
  # or 
  yarn add @salutejs/plasma-b2c@1.217.0-canary.548.5185739481.0
  yarn add @salutejs/plasma-core@1.122.0-canary.548.5185739481.0
  yarn add @salutejs/plasma-hope@1.217.0-canary.548.5185739481.0
  yarn add @salutejs/plasma-icons@1.151.0-canary.548.5185739481.0
  yarn add @salutejs/plasma-temple@1.166.0-canary.548.5185739481.0
  yarn add @salutejs/plasma-tokens-b2b@1.22.0-canary.548.5185739481.0
  yarn add @salutejs/plasma-tokens-b2c@0.31.0-canary.548.5185739481.0
  yarn add @salutejs/plasma-tokens-web@1.37.0-canary.548.5185739481.0
  yarn add @salutejs/plasma-tokens@1.56.0-canary.548.5185739481.0
  yarn add @salutejs/plasma-ui@1.198.0-canary.548.5185739481.0
  yarn add @salutejs/plasma-web@1.217.0-canary.548.5185739481.0
  yarn add @salutejs/plasma-cy-utils@0.63.0-canary.548.5185739481.0
  yarn add @salutejs/plasma-sb-utils@0.120.0-canary.548.5185739481.0
  yarn add @salutejs/plasma-tokens-utils@0.27.0-canary.548.5185739481.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-548/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-548/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-548/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-548/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-548/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-548/temple-storybook/

File: utils/plasma-tokens-native/src/generator/formatters/colorIosSwift.ts [hunk 1]
```diff
@@ -69,38 +85,76 @@ const getConditionalColorToken = (firstValue: string, secondValue: string) => `:
...
-    lightTextAccent: 'UIColor(red: 0.819, green: 0.302, blue: 0, alpha: 1)',
-    lightOnDarkTextAccent: 'UIColor(red: 0.961, green: 0.365, blue: 0.019, alpha: 1)',
-    lightSurfaceAccent: 'UIColor(red: 0.961, green: 0.365, blue: 0.019, alpha: 1)',
-    lightOnDarkSurfaceAccent: 'UIColor(red: 0.961, green: 0.365, blue: 0.019, alpha: 1)',
-    lightBannersDefaultBannerGreen: 'UIColor(red: 1, green: 0.961, blue: 0.941, alpha: 1)',
+const padFormatter = (strings: TemplateStringsArray, ...expressionValues: string[]) => {
+    const pads = expressionValues[0];
+
+    let finalString = '';
+    for (let i = 1; i < expressionValues.length; i++) {
```
Comment (Yakutoc) [inline utils/plasma-tokens-native/src/generator/formatters/colorIosSwift.ts:92]: [Уточнение]

Объясни пожалуйста в двух словах, что тут происходит?

**Review by Yakutoc — APPROVED** [review]

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

File: utils/plasma-tokens-native/src/generator/formatters/colorIosSwift.ts [hunk 2]
```diff
@@ -1,14 +1,24 @@
 import type { Dictionary, File, TransformedToken } from 'style-dictionary';
 import { upperFirstLetter, lowerFirstLetter, camelize } from '@salutejs/plasma-tokens-utils';
 
-import { GradientToken, ThemeColor, ThemeColorType } from '../../types';
+import { GradientToken, GradientType, ThemeColor, ThemeColorType } from '../../types';
 
-const getSwiftTemplate = (structContent: string, gradientContent: string, enumsContent: string) => {
+const lightTokenSberMap: Record<string, string> = {
+    lightTextAccent: 'UIColor(red: 0.819, green: 0.302, blue: 0, alpha: 1)',
```
[the remark this answers is not in the document]
Comment (neretin-trike) [reply utils/plasma-tokens-native/src/generator/formatters/colorIosSwift.ts:7]: Ага, это хардкод только для сбермаркета (он там и был)

File: utils/plasma-tokens-native/src/generator/formatters/colorIosSwift.ts (hunk 1 above)
Comment (Yeti-or) [reply utils/plasma-tokens-native/src/generator/formatters/colorIosSwift.ts:92]: отступы добавляются походу

**Review by Yeti-or — APPROVED** [review]

**Review by kayman233 — APPROVED** [review]

File: utils/plasma-tokens-native/src/generator/formatters/colorIosSwift.ts (hunk 1 above)
Comment (neretin-trike) [reply utils/plasma-tokens-native/src/generator/formatters/colorIosSwift.ts:92]: Такс, этот метод - реализация шаблонного литерала (для символов -`). Он принимает в себя два аргумента: массив строк (до переменных) и массив значений для переменных. В цикле я пробегаюсь по каждой такой строке, восстанавливаю исходный порядок и добавляю отступ, который укажет пользователь

**Pull request merged**